### PR TITLE
Introduction to common CI/CD Terms

### PR DESCRIPTION
**Pull request by matt-tyler**
WIP

Will be part of a long running series on building CI/CD on AWS

### CODE REVIEW

File: _posts/2020-04-06-cicd-part-1.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+---
+layout: post
+title: The CI/CD Codex: An Introduction to Common Terms
+date: 2020-04-06
+tags: cloud tutorial beginners aws
+author: Matthew Tyler
+image:
```
Comment (Hamo69) [inline _posts/2020-04-06-cicd-part-1.md]: may want an image of some kind..

File: _posts/2020-04-06-cicd-part-1.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+  The education most receive in software delivery is one that comes via a trial-by-fire. You start with a small solution to a small problem and iterate from there. I think it was easy to do this in years gone by when times where simpler, but the fast-paced nature of the world has become fairly unforgiving, and there are fewer low-risk opportunities to really touch-on software release processes. Increasing levels of abstraction and the proliferation of managed services make it harder to gain context and to understand the 'why', which is neccesary when faced with obstacles that you may be unfamiliar with.
+
+- To understand that software is distributed and consumed in different ways, and this inevitably influences the software development process
+
+  Software is consumed in wildly different ways depending on the who it is for. Software is installed on desktops using binaries retrieved via FTP. Libraries are distributed via various package manager. SaaS is commonly delivered via web browser. How your software is delivered and consumed has a marked effect on your build & release process.
+
+- Evaluate different CI/CD processes within the lens of the AWS Well-Architected Framework
```
Comment (Hamo69) [inline _posts/2020-04-06-cicd-part-1.md]: Are we selling a WAF?

Comment (Hamo69) [reply _posts/2020-04-06-cicd-part-1.md]: Could we link it into the new ability to provide them? Do we have a page for that?

File: _posts/2020-04-06-cicd-part-1.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+Your CD pipeline will typically be less flexible than it's CI equivalent. It will usually mandate that each stage be passed successfully before proceeding to the next.
+
+<image of a CD Process>
+
+<Add additional terms>
+
+## When Should I Introduce CI/CD Practices?
+
+<Forshaw Twitter Question>
```
Comment (Hamo69) [inline _posts/2020-04-06-cicd-part-1.md]: lol

**Comment by Hamo69** [comment]
@matt-tyler is this ready to go. Date is slotted in for the 1st June, may be worth one final glance as it looks like there are some placeholders. And the date should be updated

**Comment by Hamo69** [comment]
hey @matt-tyler did you want to kill this off or is there a plan to complete it?

**Comment by matt-tyler** [comment]
I will complete it - finding time is just difficult atm. The plan is to turn this into a series like the serverless one - it's just a little less clear to me how to organize the future installments at this stage.

The PR I just raised for the manual gates one might be a good lead-in to showing how to perform automated verification and promotion to environments, so I might make that part of this series.

I just need to get into a writing groove again.

**Comment by Hamo69** [comment]
hey @matt-tyler what shall we do with this one?

**Pull request closed without merging**